Xamarin.Forms.Platform.MacOS/NativeToolbarTracker.cs

```diff
@@ -422,18 +410,73 @@ void UpdateGroup(NativeToolbarGroup group, IList<ToolbarItem> toolbarItems, doub
 					SetAccessibility(button, element);
 					group.Items.Add(new NativeToolbarGroup.Item { ToolbarItem = item, Button = button, Element = element });
 				}
-				view.Frame = new CGRect(0, 0, totalWidth + (itemSpacing * (count - 1)), ToolbarItemHeight);
 
 				group.Group.Subitems = subItems;
 				group.Group.View = view;
+				UpdateGroupWidth(group);
 			}
 			else
 			{
 				group.Group.Subitems = new NSToolbarItem[] { };
 				group.Group.View = new NSView();
 			}
 		}
-
+		void UpdateButtonContent(NSButton button, ToolbarItem element, NativeToolbarGroup group)
+		{
+			var text = element.Text ?? "";
+			SetButtonTitle(button, text);
+			button.SizeToFit();
+			UpdateGroupWidth(group);
+			if (element.IconImageSource != null)
+			{
+				_ = element.ApplyNativeImageAsync(ToolbarItem.IconImageSourceProperty, image =>
+				{
+					if (image != null)
+					{
+						button.Image = image;
+						SetButtonTitle(button, "");
+					}
+					else
+					{
+						button.Image = null;
+					}
+					button.SizeToFit();
+					UpdateGroupWidth(group);
+				});
+			}
+			else
+			{
+				button.Image = null;
+			}
+		}
+		void UpdateGroupWidth(NativeToolbarGroup group)
+		{
+			nfloat totalWidth = 0;
+			var currentX = 0.0;
+			var view = group.Group.View;
+			for (var i = 0; i < view.Subviews.Length; i++)
+			{
+				var child = view.Subviews[i];
+				var buttonWidth = group.MinItemWidth;
+				if (child.FittingSize.Width > group.MinItemWidth)
+				{
+					buttonWidth = child.FittingSize.Width + 10;
```

Comment:
I have tried using directly `child.FittingSize.Width` and the result seems to be correct. Adding 10 to the width slightly increases the spacing, is that the goal?

Reply:
Hmm, this line was taken from existing implementation. It adds some horizontal padding when button content is long.

Padding for toolbar items with icon only is 9=44-25 (ToolbarItemWidth-ToolbarItemHeight)
I think that 10 added here to match this padding for long items.

Reply:
@jsuarezruiz I've added toolbar item with long text to ControlGallery page

Reply:
Ok, sounds good.